docs/development/pigs/pig-028.rst

```diff
@@ -0,0 +1,221 @@
+. include:: ../../references.txt
+
+.. _pig-028:
+
+***************************************
+PIG 28 - Metadata container for Gammapy
+***************************************
+
+* Author: Régis Terrier and ?
+* Created: April 14th, 2023
+* Accepted: -
+* Status: -
+* Discussion:
+
+Abstract
+========
+
+Metadata handling is crucial to correctly store information that are not directly data
+but are still required for processing, post-processing and serialization. They are
+fundamental for reproducibility.
+
+Introduction
+============
+
+As of version 1.0, Gammapy has very little support for metadata. Existing features are
+heterogeneous, hardly configurable and appear sporadically in the code, mostly in
+containers. At the DL3 level, ``EventList`` metadata is carried by its `Table.meta` dictionary.
+It is extracted from the  file FITS header which follows the GADF specifications.
+Similarly, ``Observation`` contains an `obs_info` dictionary that is build from the header as well.
+After data reduction, ``Dataset`` contains a `meta_table` which
+consists in a selection of `Observation.obs_info` entries (one table row per observation).
+During `Dataset` stacking the `meta_table` are stacked. The ``Datasets`` collection also
+agregates the `meta_table` of its members. After estimation, the ``FluxPoints`` don't
+contain any specific meta information.
+
+The algorithm classes (`Makers` and `Estimators`) don't contain any meta information so far.
+This might be an issue since some information  could be transfered on their various products
+as well.
+
+A minimal information that needs to be present on every Gammapy product and serialized
+in various formats is the `CREATOR` which is the software and software version used,
+as well as the `DATE` when the object was created, and possibly the `ORIGIN` (the user,
```

Comment:
Maybe also the data models version? ie, GADF2.0, etc? This might be useful for the DL4/DL5 products (even though we obviously dont have a formal definition yet)

Reply:
One could assume that the internal data format can be deduced from the gammapy version. I don't know whether we want to introduce an additional versioning.
Regarding serialization data formats, I think the IO code should be the place where the correct keywords are defined.